core/trino-spi/src/main/java/io/trino/spi/statistics/StatsUtil.java

```diff
@@ -85,4 +85,44 @@ public static OptionalDouble toStatsRepresentation(Type type, Object value)
 
         return OptionalDouble.empty();
     }
+
+    public static boolean areStatsRepresentationsCompatible(Type type, Type otherType)
+    {
+        requireNonNull(type, "type is null");
+        requireNonNull(otherType, "otherType is null");
+
+        if (type.equals(otherType)) {
+            return true;
+        }
+
+        if (type == BOOLEAN || type == REAL || type == DOUBLE || type == DATE) {
+            return false;
+        }
+        if (type == TINYINT || type == SMALLINT || type == INTEGER || type == BIGINT) {
+            return (otherType == TINYINT || otherType == SMALLINT || otherType == INTEGER || otherType == BIGINT);
```

Comment:
braces are redundant.

Reply:
also, `decimal(p, 0)` surely has compatible stats representation with integer types (may matter for Oracle connector, for example)

also, `decimal(p, s)` has "compatible" stats representation, depending on the definition of "compatible"

Reply:
I've added compatibility for `decimal(p, 0)` with integer types and removed extra braces